aten/src/ATen/native/native_functions.yaml

```diff
@@ -1923,6 +1923,10 @@
     CUDA: batch_norm_cuda
     MkldnnCPU: mkldnn_batch_norm
 
+- func: native_batch_norm.out(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps, *, Tensor(a!) out) -> (Tensor(a!), Tensor, Tensor)
```

Comment:
I think the out variant has to take out values for EVERY return value. So you can't just take `out`, you also have to do the other two tensors. Look at kthvalue.dimname_out for an example.

Reply:
running_mean / running_var are optional, making them as output values may require more code modifications around the calls to native_batch_norm...

do you think it makes more sense to figure it out (I'll see closer if many modifications are needed) or instead to fix codegen?

Reply:
I'm not sure. The thing you are trying to do here is very strange and I'm not sure if we actually support it directly.

Reply:
It's also

Reply:
Partly it's because the current approach is to return some intermediate computation from batch_norm (mean, invstd): https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/Normalization.cuh#L540-L546

For standard version I could allocate these tensors outside the out-function. Can dispatch be asked to pass in a reference to a stub empty Tensor (to be resized inside the out-function)?